### PR TITLE
allow to leave missing signatures as null

### PR DESCRIPTION
**Pull request by wiyarmir**
this helps backend do partial signatures in the right places

we also ensure fee payer stays at the top

### CODE REVIEW

File: solana-kotlin/src/commonMain/kotlin/net/avianlabs/solana/domain/core/AccountKeysList.kt [hunk 1]
```diff
@@ -1,6 +1,10 @@
 package net.avianlabs.solana.domain.core
 
-internal fun List<AccountMeta>.normalize(): List<AccountMeta> = groupBy { it.publicKey }
+import net.avianlabs.solana.tweetnacl.ed25519.PublicKey
+
+internal fun List<AccountMeta>.normalize(
```
Comment (jszmltr) [inline solana-kotlin/src/commonMain/kotlin/net/avianlabs/solana/domain/core/AccountKeysList.kt:5]: would it make sense to have two separate functions ? 
`normalize()` and `normalizeWithFeePayer` ?

Comment (wiyarmir) [reply solana-kotlin/src/commonMain/kotlin/net/avianlabs/solana/domain/core/AccountKeysList.kt:5]: no, fee payer might be set or not

File: solana-kotlin/src/commonMain/kotlin/net/avianlabs/solana/domain/core/Transaction.kt [hunk 2]
```diff
@@ -26,13 +22,9 @@ public open class Transaction internal constructor(
...
-    val signatures = signers.map { signer ->
-      TweetNaCl.Signature.sign(serializedMessage, signer.secretKey).encodeToBase58String()
-    }
-
-    val signatureSet = signatures.toSet()
-    if (signatureSet.size != signatures.size) {
-      logger.warn { "Duplicate signatures detected" }
+    val signatures = signers.associate { signer ->
+      signer.publicKey to
+        TweetNaCl.Signature.sign(serializedMessage, signer.secretKey)
```
Comment (jszmltr) [inline solana-kotlin/src/commonMain/kotlin/net/avianlabs/solana/domain/core/Transaction.kt:27]: Why do we do this now ?

**Review by jszmltr — APPROVED** [review]

File: solana-kotlin/src/commonMain/kotlin/net/avianlabs/solana/domain/core/Transaction.kt (hunk 2 above)
Comment (wiyarmir) [reply solana-kotlin/src/commonMain/kotlin/net/avianlabs/solana/domain/core/Transaction.kt:27]: so the signatures can preserve the order of account meta list

**Pull request merged**